internal/ingress/controller/nginx.go

```diff
@@ -829,6 +830,7 @@ func configureDynamically(pcfg *ingress.Configuration, port int, isDynamicCertif
 			Name:      key,
 			Endpoints: ep.Endpoints,
 			Port:      intstr.FromInt(ep.Port),
+			Service:   ep.Service,
```

Comment:
Please check https://github.com/kubernetes/ingress-nginx/pull/3615/files#diff-cde3fffe2425ad7efaa8add1d05ae2c0R784. We should not send a full service object.